scripts/deploy-production.js

```diff
@@ -14,113 +22,89 @@ const main = async () => {
       type: "confirm",
       name: "value",
-      message: "No new commits. Do you want to deploy anyway?",
+      message:
+        "No new commits. Do you want to deploy the most recent existing tag?",
       initial: false,
     });
-    shouldDeployLocal = response.value;
-    if (!shouldDeployLocal) {
+
+    const forceDeploy = response.value;
+    if (forceDeploy) {
+      await deleteAndPushLatestTag();
       return;
     }
-  } else {
-    console.log("Commit messages:");
-    console.log(commitMessages);
+
+    console.log(
+      "Deploy canceled. If systems are experiencing an outage and you'd like to push the deploy directly to S3, run yarn deploy:prod --local."
+    );
+    return;
   }
 
+  // Print all commits between the last tag and the current commit
+  console.log("Commit messages:\n" + commitMessages);
+
   const response = await prompts({
     type: "confirm",
     name: "value",
     message: "Are you sure you want to deploy to production?",
   });
 
   if (response.value) {
-    if (shouldDeployLocal) {
-      try {
-        const localDeployOutput = await runLocalDeploy();
-        console.log(localDeployOutput);
-      } catch (err) {
-        console.log(err);
-        console.log("Local deploy failed. Aborting.");
-      }
-    } else {
-      try {
-        const createTagOutput = await createNewTag();
-        console.log(createTagOutput);
-        console.log("Pushed to remote. Should be deploying soon...");
-        console.log(
-          "Track deploy progress at https://spruce.mongodb.com/commits/spruce?requester=git_tag_request"
-        );
-      } catch (err) {
-        console.log(err);
-        console.log("Creating tag failed. Aborting.");
-      }
+    try {
+      const createTagOutput = await createNewTag();
+      console.log(createTagOutput);
+      console.log("Pushed to remote. Should be deploying soon...");
+      console.log(
+        "Track deploy progress at https://spruce.mongodb.com/commits/spruce?requester=git_tag_request"
+      );
+    } catch (err) {
+      console.log(err);
+      console.log("Creating tag failed. Aborting.");
     }
   }
-};
 
-const createNewTag = () =>
-  new Promise((resolve, reject) => {
-    exec("yarn version --new-version patch", (err, stdout) => {
-      if (err) {
-        reject(err);
-        return;
-      }
-      resolve(stdout);
-    });
-  });
+  return;
+};
 
-const getCommitMessages = () =>
-  new Promise((resolve, reject) => {
-    exec(
-      "git log $(git describe --tags --abbrev=0)..HEAD --oneline",
-      (err, stdout) => {
-        if (err) {
-          reject(err);
-          return;
-        }
-        resolve(stdout);
-      }
-    );
+/* Deploy by generating a production build locally and pushing it directly to S3. */
+const localDeploy = async () => {
+  const response = await prompts({
+    type: "confirm",
+    name: "value",
+    message:
+      "Are you sure you'd like to build Spruce locally and push directly to S3?",
```

Comment:
Can we add some disclaimers here about how this is a dangerous operation? 

I want to ensure whoever is running this knows about the risks since many things can go wrong.
Such as deploying outdated deps,
outdated env variables,
deploying untested code,